contracts/src/arbitration/IArbitrableV2.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8;
+
+import "./IArbitratorV2.sol";
+
+/**
+ * @title IArbitrableV2
+ * Arbitrable interface.
+ * When developing arbitrable contracts, we need to:
+ * - Define the action taken when a ruling is received by the contract.
+ * - Allow dispute creation. For this a function must call arbitrator.createDispute{value: _fee}(_choices,_extraData);
+ */
+interface IArbitrableV2 {
+    /**
+     * @dev To be emitted when a dispute is created to link the correct meta-evidence to the disputeID.
+     * @param _arbitrator The arbitrator of the contract.
+     * @param _disputeID ID of the dispute in the Arbitrator contract.
+     * @param _externalDisputeID Unique identifier from the dispute creator that is linked to this dispute.
+     * @param _disputeContextUri IPFS path to dispute context, example: '/ipfs/Qmarwkf7C9RuzDEJNnarT3WZ7kem5bk8DZAzx78acJjMFH/disputecontext.json'
+     */
+    event Dispute(
+        IArbitrableV2 indexed _arbitrator,
+        uint256 indexed _disputeID,
+        uint256 _externalDisputeID,
+        string _disputeContextUri
```

Comment:
In defense of `MetaEvidence` event (or equivalent `DisputeContext`), if you emit this separately, and then you reference it with an ID, that's much cheaper.

What should I assume the arbitrator do in this instance, store the `_disputeContextUri` in storage? Then every challenge will require loading this string from storage (which could be ~6 cold SLOAD or 12k gas when making a challenge).

If you instead were to have the event and you just use an ID, then, that's just 1 cold SLOAD.